llvm/include/llvm/IR/InstrTypes.h

```diff
@@ -2199,8 +2199,8 @@ class CallBase : public Instruction {
   FPClassTest getParamNoFPClass(unsigned i) const;
 
   /// If this return value has a range attribute, return the value range of the
-  /// argument. Otherwise, std::nullopt is returned.
-  std::optional<ConstantRange> getRange() const;
+  /// argument. Otherwise, nullptr is returned.
+  const ConstantRange *getRange() const;
```

Comment:
I'm not sure about this one -- e.g. if we wanted to intersect the range on the call and the declaration, we'd no longer be able to do it with this API...

Reply:
I'm not sure that I understand this comment, the intersect methods seems to works on `const ConstantRange*` as it is e.g.
`ConstantRange intersectWith(const ConstantRange &CR, PreferredRangeType Type = Smallest) const;` 
or was there something else that you was thinking about?

Reply:
You would not be able to return the result of the intersection in a const ConstantRange * type.

Reply:
ok you was thinking that the intersection was done in the body of this function. But is it interesting to return the intersection shall not only the smallest of the ranges be returned feels like some thing is wrong if the the call do not contain the same range or a subrange of the declaration

Reply:
so can call the `bool ConstantRange::contains(const ConstantRange &Other) const` to see what pointer to return

Reply:
hmm or after https://github.com/llvm/llvm-project/pull/90134 it maybe is not possible to say that the call is a subrange of the declaration anymore